kernel/string/itoa.c

```diff
@@ -5,12 +5,9 @@
  * See the file "LICENSE" in the source distribution for details *
  *****************************************************************/
 
-#ifndef _ITOA_C
-#define _ITOA_C
-
 #include <types.h>
 
-void uitoa3(uint64_t n, char *s, uint8_t len, uint8_t base){
+void uitoa3(uint64_t n, char *s, uint16_t len, uint8_t base){
```

Comment:
If len is changed from uint8_t to uint16_t, then the index variable (i) for iterating over s needs to be changed accordingly or it won't work.

Reply:
fix, thanks

Reply:
Now that I look at it again...there shouldn't be a need for a uint16_t.  This function gets called to convert an integer (n) into a character string (s, which is passed by the caller as a pointer and is effectively the return value).  There is no circumstance in which a 64-bit unsigned integer-to-string conversion would result in a string longer than 65 bytes + \0, so unless there's something I'm not considering it seems like a uint8_t should be sufficient for all cases, no?  Even if the caller passes a pointer to an array longer than 256 bytes, this function will never have any reason to write beyond s[65].

On the other hand, i and len are all on the stack, so they're both essentially taking up 64 bits each regardless under the hood whether they're uint8 or uint16 and it's not actually making the kernel image larger, so it's probably not worth worrying about (unless, again, there's something I'm not thinking of).